drivers/adc/ad738x/iio_ad738x.c

```diff
@@ -0,0 +1,207 @@
+/***************************************************************************//**
+* @file   iio_ad738x.c
+* @brief  Implementation of AD738X IIO interface
+********************************************************************************
+* Copyright (c) 2023 Analog Devices, Inc.
+* Copyright (c) 2023 BayLibre, SAS.
+* All rights reserved.
+*
+* This software is proprietary to Analog Devices, Inc. and its licensors.
+* By using this software you agree to the terms of the associated
+* Analog Devices Software License Agreement.
+*******************************************************************************/
+#ifdef IIO_SUPPORT
+
+#include <stdio.h>
+#include <math.h>
+
+#include "ad738x.h"
+#include "iio_ad738x.h"
+#include "iio.h"
+#include "no_os_util.h"
+#include "no_os_alloc.h"
+#include "no_os_error.h"
+
+#define AD738X_CH(_idx) {\
+	.name = "voltage" #_idx, \
+	.ch_type = IIO_VOLTAGE,\
+	.ch_out = IIO_DIRECTION_INPUT,\
+	.indexed = true,\
+	.channel = _idx,\
+	.scan_index = _idx,\
+	.scan_type = &ad738x_iio_scan_type,\
+	.attributes = ad738x_iio_ch_attributes,\
+}
+
+#define CHN_STORAGE_BITS	16
+#define ADC_RESOLUTION		16
+#define ADC_REF_VOLTAGE		3300
+#define ADC_MAX_COUNT_UNIPOLAR  (uint32_t)((1 << ADC_RESOLUTION) - 1)
+
+struct ad738x_iio {
+	struct ad738x_dev *drv_dev;
+	struct iio_device *iio_dev;
+};
+
+/**
+ * @brief       Read the debug register value
+ * @param       dev[in, out]- Pointer to IIO device instance
+ * @param       reg[in]- Register address to read from
+ * @param       readval[out]- Pointer to variable to read data into
+ * @return      0 in case of success, negative value otherwise
+ */
+static int32_t iio_ad738x_debug_reg_read(void *device, uint32_t reg,
+		uint32_t *val)
+{
+	struct ad738x_iio_dev *iio_dev = device;
+	struct ad738x_dev *dev = iio_dev->ad738x_dev;
+
+	return ad738x_spi_reg_read(dev,  (uint8_t) reg, (uint16_t *)val);
+}
+
+/**
+ * @brief       Write the debug register value
+ * @param       dev[in, out]- Pointer to IIO device instance
+ * @param       reg[in]- Register address to read from
+ * @param       writeval[out]- Pointer to variable to write data into
+ * @return      0 in case of success, negative value otherwise
+ */
+static int32_t iio_ad738x_debug_reg_write(void *device, uint32_t reg,
+		uint32_t val)
+{
+	struct ad738x_iio_dev *iio_dev = device;
+	struct ad738x_dev *dev = iio_dev->ad738x_dev;
+
+	return ad738x_spi_reg_write(dev, (uint8_t)reg, (uint16_t)val);
+}
+
+static int get_raw(void *device,
+		   char *buf,
+		   uint32_t len,
+		   const struct iio_ch_info *channel,
+		   intptr_t id)
```

Comment:
no need to add each parameter on a new line, just write the parameters on the same line and let astyle take care of the formatting for you. see everywhere where it applies.